app/controllers/application_controller.rb

```diff
@@ -2,4 +2,20 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  helper_method :current_user
+
+  before_action :ensure_current_user, only: [:new, :create, :update, :destroy]
+
+  def logged_in?
```

Comment:
I don't think you use this anywhere? In any case, I'd probably just ditch is and use "if current_user" anywhere you would use "if logged_in?"